django/cantusdb_project/main_app/templates/chant_detail.html

```diff
@@ -215,6 +215,14 @@ <h3>{{ chant.incipit }}</h3>
                         <a href="{{ chant.image_link }}" target="_blank" style="word-break: break-all">{{ chant.image_link }}</a>
                     </dd>
                 {% endif %}
+                {% if chant.source.exists_on_cantus_ultimus %}
```

Comment:
Remember that traversing multiple models in a template is ~~~the path to the dark side~~~ a surefire way to explode the number of queries that are performed on a page load. Make sure you test this!

Reply:
Makes sense. I can't exactly test how many queries are being made because of #1457, but I assume passing this as a `context` template variable in the view would be better than accessing it through the template.

Reply:
Or, even better, checking your `queryset` in your view before it goes to the template.

Reply:
True, right now our `ChantDetailView` isn't using a `queryset` attribute and everything is being passed in as context data. If I'm understanding correctly, this is probably part of a larger issue where we could clean up and optimize our code by using `get_queryset` where possible.